CRM/Contribute/Page/UserDashboard.php

```diff
@@ -163,11 +133,68 @@ public function isIncludeInvoiceLinks() {
   /**
    * the main function that is called when the page
    * loads, it decides the which action has to be taken for the page.
+   *
+   * @throws \CRM_Core_Exception
    */
   public function run() {
     $this->assign('isIncludeInvoiceLinks', $this->isIncludeInvoiceLinks());
     parent::preProcess();
     $this->listContribution();
   }
 
+  /**
+   * Get the contact's contributions.
+   *
+   * @param bool $isSoftCredit
+   *   Return contributions for which the contact is the soft credit contact instead.
+   *
+   * @return array
+   * @throws \CRM_Core_Exception
+   */
+  protected function getContributions(bool $isSoftCredit = FALSE): array {
+    $apiQuery = Contribution::get(FALSE)
+      ->addOrderBy('receive_date', 'DESC')
+      ->setLimit(12)
+      ->setSelect([
+        'total_amount',
+        'contribution_recur_id',
+        'receive_date',
+        'receipt_date',
+        'cancel_date',
+        'amount_level',
+        'contact_id',
+        'contact_id.display_name',
+        'contribution_status_id:name',
+        'contribution_status_id:label',
+        'financial_type_id:label',
+        'currency',
+        'amount_level',
+        'contact_id,',
+        'source',
+        'balance_amount',
```

Comment:
Just noting this is ok because it is a computed field https://github.com/civicrm/civicrm-core/blob/5d378881aa065a348cc2952c5393814a82e38474/Civi/Api4/Service/Spec/Provider/ContributionGetSpecProvider.php#L50